testing_and_setup/compass/ocean/scripts/plots/plot_initial_state.py

```diff
@@ -51,63 +52,70 @@ def main():
 
     plt.subplot(3, 3, 2)
     varName = 'maxLevelCell'
-    var = ncfile.variables[varName]
-    maxLevelCell = var[:]
-    plt.hist(var, bins=nVertLevels - 4)
+    var = ds[varName]
+    maxLevelCell = var.values - 1
+    xarray.plot.hist(var, bins=nVertLevels - 4)
     plt.ylabel('frequency')
     plt.xlabel(varName)
-    txt = '{}{:9.2e} {:9.2e} {}\n'.format(txt, np.amin(var), np.amax(var), varName)
+    txt = '{}{:9.2e} {:9.2e} {}\n'.format(txt, var.min().values, 
+                                          var.max().values, varName)
 
     plt.subplot(3, 3, 3)
     varName = 'bottomDepth'
-    var = ncfile.variables[varName]
-    plt.hist(var, bins=nVertLevels - 4)
+    var = ds[varName]
+    xarray.plot.hist(var, bins=nVertLevels - 4)
     plt.xlabel(varName)
-    txt = '{}{:9.2e} {:9.2e} {}\n'.format(txt, np.amin(var), np.amax(var), varName)
+    txt = '{}{:9.2e} {:9.2e} {}\n'.format(txt, var.min().values, 
+                                          var.max().values, varName)
 
-    cellsOnEdge = ncfile.variables['cellsOnEdge']
+    cellsOnEdge = ds['cellsOnEdge'].values - 1
     cellMask = np.zeros((nCells, nVertLevels), bool)
     edgeMask = np.zeros((nEdges, nVertLevels), bool)
     for k in range(nVertLevels):
-        cellMask[:, k] = k < maxLevelCell
-        cell0 = cellsOnEdge[:, 0]-1
-        cell1 = cellsOnEdge[:, 1]-1
+        cellMask[:, k] = k <= maxLevelCell
+        cell0 = cellsOnEdge[:, 0]
+        cell1 = cellsOnEdge[:, 1]
         edgeMask[:, k] = np.logical_and(np.logical_and(cellMask[cell0, k],
                                                        cellMask[cell1, k]),
                                         np.logical_and(cell0 >= 0,
                                                        cell1 >= 0))
+    cellMask = xarray.DataArray(data=cellMask, dims=('nCells', 'nVertLevels'))
+    edgeMask = xarray.DataArray(data=edgeMask, dims=('nEdges', 'nVertLevels'))
```

Comment:
By making these masks into `xarray.DataArray`s, we can use them to mask other data arrays with `where()`.